gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -671,12 +674,29 @@ public FSDataOutputStream createNonRecursive(
       throws IOException {
     URI gcsPath = getGcsPath(checkNotNull(hadoopPath, "hadoopPath must not be null"));
     URI parentGcsPath = UriPaths.getParentPath(gcsPath);
+    DurationTracker get_tracker = getDurationTrackerFactory().trackDuration(GHFSStatistic.ACTION_HTTP_GET_REQUEST.getSymbol());
+    DurationTracker head_tracker = getDurationTrackerFactory().trackDuration(GHFSStatistic.ACTION_HTTP_HEAD_REQUEST.getSymbol());
+
```

Comment:
This is not where HTTP requests are initiated, we need to integrate on much-lover level in HTTP client library to know when actually HTTP requests are sent and to be able to measure their duration.

Reply:
@medb HTTP requests are initiated from gcsio. To track the duration from gcsio, we need to create a instrumentation in gcsio which also includes creating the duration tracker factory and relevant classes.

Reply:
Yes, this is exactly what we should do - there are no other way to get HTTP requests statistic.

Reply:
Updated